### PR TITLE
fix(installWizard): Fix client default values not set 

### LINKED ISSUE #12813: DSN is empty
## Important Details

How are you running Sentry?

* [ ] On-Premise docker 
* [ ] sentry version 9.1.0 - onbuild
* [ ] docker-compose 

## Description

image

### PR DESCRIPTION
**Pull request by BYK**
Fixes #12813.

Just checked the payload to the server after the fix, it is now as follows w/ onpremise:

```
{
    "auth.allow-registration": false,
    "beacon.anonymous": false,
    "mail.from": "root@localhost",
    "mail.password": "lol really",
    "mail.port": 25,
    "mail.use-tls": false,
    "mail.username": "root",
    "system.admin-email": "ben@byk.im",
    "system.url-prefix": "http://localhost:9000"
}
```

This also indeed fixes the empty DSN issue.

### CODE REVIEW

**Review by dcramer — APPROVED** [review]
I will trust your tests on this. The forms/options behavior here was super problematic to work around. I think we should talk with @billyvg / @lynnagara about removing our duplicate form components and maybe standardizing using Formik.

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-8.15.1
+8.16.0
```
Comment (billyvg) [inline .nvmrc]: Is this intentional?

File: src/sentry/static/sentry/app/options.jsx [hunk 2]
```diff
@@ -183,8 +188,8 @@ export function getOptionField(option, field) {
       {...meta}
       name={option}
       key={option}
-      defaultValue={meta.defaultValue ? meta.defaultValue() : undefined}
-      required={meta.required && !meta.allowEmpty}
+      defaultValue={getOptionDefault(option)}
+      required={meta.required && !meta.allowEmpt}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/options.jsx]: typo?

**Review by billyvg — CHANGES_REQUESTED** [review]

File: .nvmrc (hunk 1 above)
Comment (BYK) [reply .nvmrc]: nope, happened during testing.

File: src/sentry/static/sentry/app/options.jsx (hunk 2 above)
Comment (BYK) [reply src/sentry/static/sentry/app/options.jsx]: yup 😢

File: tests/js/spec/views/installWizard.spec.jsx [hunk 3]
```diff
@@ -80,14 +80,14 @@ describe('InstallWizard', function() {
 
     expect(
       wrapper.find('input[name="beacon.anonymous"][value="false"]').prop('checked')
-    ).toBe(true);
+    ).toBe(false);
```
Comment (billyvg) [inline tests/js/spec/views/installWizard.spec.jsx]: Since `beacon.anonymous.value` is `false`, shouldn't this be checked?

Comment (BYK) [reply tests/js/spec/views/installWizard.spec.jsx]: Nope, see https://github.com/getsentry/sentry/blob/606c95547ce11d4a85f1279a8a699041fd31fdf8/src/sentry/static/sentry/app/views/installWizard.jsx#L67-L68

Comment (billyvg) [reply tests/js/spec/views/installWizard.spec.jsx]: Or is this intentional where regardless of the value, both options are always unchecked

**Review by billyvg — APPROVED** [review]

File: tests/js/spec/views/installWizard.spec.jsx (hunk 3 above)
Comment (BYK) [reply tests/js/spec/views/installWizard.spec.jsx]: Yeah, it is intentional. Not sure why this test didn't fail earlier. Need to look into all this random CI stuff soon.

**Pull request merged**